src/build/app/mod.rs

```diff
@@ -784,6 +786,28 @@ impl<'b> App<'b> {
         self
     }
 
+    /// Replaces an argument to this app with other arguments.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::{App, Arg, };
+    /// let m = App::new("app")
+    ///     .replacer("install", &["module", "install"])
+    ///     .subcommand(App::new("module")
+    ///         .subcommand(App::new("install")))
+    ///     .get_matches_from(vec!["app", "install"]);
+    ///
+    /// assert!(m.subcommand_matches("module").is_some());
+    /// assert!(m.subcommand_matches("module").unwrap().subcommand_matches("install").is_some());
+    /// ```
+    ///
+    /// Here, when args given are `app install`, they will be changed to `app module install`.
+    pub fn replacer(mut self, name: &'b str, target: &'b [&'b str]) -> Self {
```

Comment:
`replacer` is not a good name here. `delegate` would fit much better in my opinion.

Also, the doc comment should be much more descriptive.

Reply:
No, `delegate` does not fit here. It was the first name I thought of. I could change it to `replace` to make it better. Or `substitute`.

I thought the documentation was pretty self explanatory. What am I missing?

Reply:
`replace` explains _the way it's implemented_ while `delegate` explains _the way it's supposed to be used_. I think users are interested in the latter more.

The same problem with docs: this comment describes the way it's implemented but, it my opinion, it should explain why this is useful and what it can be used for. It should also talk about the specific use case outlined in the linked issue.

Reply:
`replace` might be wrong, but `delegate` for sure doesn't fit here. 

> entrust (a task or responsibility) to another

We are not actually doing that.

Reply:
OK, let's just rename it to `replace` because all the other methods use verbs or adverbs.